mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java

```diff
@@ -0,0 +1,83 @@
+import uk.nhs.adaptors.mockmhsservice.producer.InboundProducer;
+
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class MockMhsService {
+
+    private final ObjectMapper objectMapper;
+    private final InboundProducer inboundProducer;
+
+    public ResponseEntity<String> handleRequest(
+            String interactionId,
+            String waitForResponse,
+            String fromAsid,
+            String messageId,
+            String correlationId,
+            String odsCode,
+            String mockMhsMessage) throws IOException {
+
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, true);
+        objectMapper.configure(DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES, true);
+
+        ObjectNode rootNode = objectMapper.createObjectNode();
+
+        String responseJsonString;
+        String mockSuccessMessage = "Message acknowledged.";
+        String mockInteractionIdErrorMessage = "Error, cannot handle request header Interaction-Id.";
+        String mockRequestBodyErrorMessage = "Error, content of request body does not match expected JSON";
+        String mockInboundReplyErrorMessage = "Error, could not produce inbound reply.";
+
+        String mockValidInteractionId = "RCMR_IN030000UK06";
+
+        try {
+            verifyJson(mockMhsMessage);
+        } catch (JsonProcessingException e) {
+            rootNode.put("message", mockRequestBodyErrorMessage);
+            responseJsonString = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(rootNode);
+            return new ResponseEntity<>(responseJsonString, INTERNAL_SERVER_ERROR);
+        }
+
+        if (interactionId.equals(mockValidInteractionId)) {
+            try {
+                var inputStream = this.getClass().getClassLoader().getResourceAsStream("COPC_IN000001UK01.xml");
+                var text = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+
+                inboundProducer.sendToMhsInboundQueue(text);
+                rootNode.put("message", mockSuccessMessage);
+                responseJsonString = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(rootNode);
+                return new ResponseEntity<>(responseJsonString, ACCEPTED);
+            } catch (JmsException e) {
+                rootNode.put("message", mockInboundReplyErrorMessage);
+                responseJsonString = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(rootNode);
+                return new ResponseEntity<>(responseJsonString, INTERNAL_SERVER_ERROR);
+            }
+        }
+
+        rootNode.put("message", mockInteractionIdErrorMessage);
+        responseJsonString = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(rootNode);
+        return new ResponseEntity<>(responseJsonString, INTERNAL_SERVER_ERROR);
+    }
+
+    private void verifyJson(String json) throws JsonProcessingException {
```

Comment:
I'd rename this method to something like getMessagePayload since you're returning a value